### PR TITLE
Testing new setup

### PR DESCRIPTION
**Pull request by matq007**
A new approach where each tool has its own container. For now, I'm using my dev image because I've modified the main Dockerfile. If the tests pass I will make another change where I will rename the image so it can be built properly (we have a hook setup on the dev branch to build the Dockefile).

### CODE REVIEW

**Comment by matq007** [comment]
I've managed to setup travis on my repo, so I'll come back to this one in a moment.

**Comment by matq007** [comment]
This is the current setup where each tool has its own container. The structure enables to create tests for each tool, but due to timeouts and seg faults (running out of RAM) only star-fusion works for now. I think when we minimize the test read we might be able to run the rest of the tests.
I've made a small change in `travis.yml` so that for the dev branch is using `nfcore/rnafusion:dev`. I've included build scripts to allow some automation for us.

**Comment by MaxUlysse** [comment]
Small issue with Travis CI

File: .travis.yml [hunk 1]
```diff
@@ -9,11 +9,14 @@ matrix:
 before_install:
   # PRs made to 'master' branch should always orginate from another repo or the 'dev' branch
   - '[ $TRAVIS_PULL_REQUEST = "false" ] || [ $TRAVIS_BRANCH != "master" ] || ([ $TRAVIS_PULL_REQUEST_SLUG = $TRAVIS_REPO_SLUG ] && [ $TRAVIS_PULL_REQUEST_BRANCH = "dev" ])'
-  # Pull the docker image first so the test doesn't wait for this
-  - docker pull nfcore/rnafusion
-  # Fake the tag locally so that the pipeline runs properly
-  - docker tag nfcore/rnafusion nfcore/rnafusion:latest
-  # Pull the docker image first so the test doesn't wait for this
+  - if [ $TRAVIS_BRANCH == "master" ]; then 
+      docker pull nfcore/rnafusion;
```
Comment (MaxUlysse) [inline .travis.yml]: Shouldn't you pull nfcore/rnafusion:latest each time?

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "/usr/local/miniconda2/bin/python"
```
Comment (MaxUlysse) [inline .vscode/settings.json]: Do we really need this file in the repo?

File: conf/docker.config [hunk 3]
```diff
@@ -16,6 +16,16 @@ docker {
 }
 
 process {
-  container = wf_container
+  container = 'nfcore/rnafusion:dev'
+
+  withName:star_fusion {
+    container = 'nfcore/rnafusion:star-fusion_v1.4.0'
```
Comment (MaxUlysse) [inline conf/docker.config]: I can see that you tagged each tool with a different version.
I'm guessing it's the version of the tool.
Shouldn't it be easier to have a single tag for all tool, like the release from this pipeline for example?

File: conf/uppmax-devel.config [hunk 4]
```diff
@@ -14,6 +14,17 @@ executor {
 }
 process {
   queue = 'devel'
+  container = 'file:///home/matq007/private/pipeline_images/rnafusion-dev.simg'
```
Comment (MaxUlysse) [inline conf/uppmax-devel.config]: You can add a params like containerPath.
cf: https://github.com/SciLifeLab/Sarek/blob/e504b0e7c1f025641469b76edbc64e4a04bdb009/conf/singularity-path.config

File: main.nf [hunk 5]
```diff
@@ -219,10 +232,11 @@ ${summary.collect { k,v -> "            <dt>$k</dt><dd><samp>${v ?: '<span style
  */
 process star_fusion {
     tag "$name"
+    conda ''
```
Comment (MaxUlysse) [inline main.nf]: Is this useful?

File: main.nf [hunk 6]
```diff
@@ -321,16 +338,19 @@ process fusion_inspector_preprocess {
 /*
  * Fusion Inspector
  */
+if (params.test) {
+    fusions = Channel.fromPath("$baseDir/tools/fusion-inspector/fusions.txt")
```
Comment (MaxUlysse) [inline main.nf]: Need to check if this work when executing:
```nextflow run nf-core/rnafusion ...```

File: .travis.yml (hunk 1 above)
Comment (matq007) [reply .travis.yml]: By default if the tag is not specified, latest will be pulled. I've just copy pasted the required lines by `nf-core lint`:
ERROR: Test Failures:
  http://nf-co.re/errors#5: CI is not pulling the correct docker image. Should be:
    'docker pull nfcore/rnafusion'
  http://nf-co.re/errors#5: CI is not tagging docker image correctly. Should be:
    'docker tag nfcore/rnafusion nfcore/rnafusion:latest'

File: .vscode/settings.json (hunk 2 above)
Comment (matq007) [reply .vscode/settings.json]: This should not be here, you are right!

File: conf/docker.config (hunk 3 above)
Comment (matq007) [reply conf/docker.config]: That would be a good idea, the problem is that some of the tools don't support `--version` parameter so you have no idea what version you are running. In this case, the `environment.yml` saves the day but for example, for `fusioncatcher` there is just commit message with a version. Based on this chose this approach instead. I think tagging it by pipeline release version can be confusing afterward.

File: conf/uppmax-devel.config (hunk 4 above)
Comment (matq007) [reply conf/uppmax-devel.config]: Ah, this is much better solution, thank you. I'll include this one 👍

File: main.nf (hunk 5 above)
Comment (matq007) [reply main.nf]: Ah, I must have forgotten about that. I was testing something with conda. I will remove it.

File: main.nf (hunk 6 above)
Comment (matq007) [reply main.nf]: I'll run the pipeline one more time on UPPMAX but the pipeline worked as expected. I'll come back to this tomorrow after running the pipeline.

Comment (matq007) [reply main.nf]: Works as expected.

File: conf/docker.config (hunk 3 above)
Comment (matq007) [reply conf/docker.config]: Alright, moving to proposed new semantic versioning. New convention is nfcore/rnafusion:<tool>_v<pipeline_version>

File: .travis.yml (hunk 1 above)
Comment (matq007) [reply .travis.yml]: After discussion, this doesn't solve the problem. Dropping down to
```
- docker pull nfcore/rnafusion:dev
- docker tag nfcore/rnafusion:dev nfcore/rnafusion:latest
- docker pull nfcore/test-datasets:rnafusion
```

**Pull request merged**